### PR TITLE
updating docs and removing BuildConfiguration and ConfigurationGroup

### PR DESCRIPTION
**Pull request by Anipik**
BuildConfiguration will no longer is a global property passed to traversal projects.
BuildCOnfiguration -> BuildSettings. It is just used as a string concatenation or kind of shorter representation of BuildTargetFramework-OSGroup-COnfiguration-ArchGroup.

Developers shouldn't use this this property for anything related to building the repo or individual projects.

The pr also updates are docs for the new config system

### CODE REVIEW

File: docs/coding-guidelines/adding-api-guidelines.md [hunk 1]
```diff
@@ -24,20 +24,20 @@ the implementation without compat concerns in future releases.
 
 ### Determine target framework
 
-`netstandard` or `netcoreapp` is the target framework version currently under development.
+`netstandard2.1` or `netcoreapp5.0` is the target framework version currently under development.
```
Comment (ViktorHofer) [inline docs/coding-guidelines/adding-api-guidelines.md]: `netstandard2.1` is already released, it's not under development.

Comment (ViktorHofer) [reply docs/coding-guidelines/adding-api-guidelines.md]: I think we can remove most of the netstandard mentions here as going forward net5 (netcoreapp5.0) should replace all these considerations.

File: docs/coding-guidelines/adding-api-guidelines.md [hunk 2]
```diff
@@ -47,19 +47,16 @@ the implementation without compat concerns in future releases.
...
   - If assembly or package version is updated the package index needs to be updated by running
     `dotnet msbuild <Library>/pkg/<Library>.pkgproj /t:UpdatePackageIndex`
 
 **Update tests**
-  - Set `TargetGroup` which will generally match the `TargetGroup` in the src library build configuration. (ex: [System.Runtime\tests\Configurations.props](https://github.com/dotnet/corefx/blob/master/src/System.Runtime/tests/Configurations.props#L3))
+  - Add new `TargetFramework` to the ```TargetFrameworks```.
   - Add new test code following [conventions](project-guidelines.md#code-file-naming-conventions) for new files to that are specific to the new target framework.
-  - To run just the new test configuration run `dotnet msbuild <Library>.csproj /t:RebuildAndTest /p:TargetGroup=<TargetGroup>`
+  - To run just the new test targetFramework run `dotnet msbuild <Library>.csproj /t:RebuildAndTest /p:BuildTargetFramework=<TargetFramework>`
```
Comment (ViktorHofer) [inline docs/coding-guidelines/adding-api-guidelines.md]: @Anipik @ericstj so dotnet build -f net472 /t:RebuildAndTest doesn't work with the new config system? What is necessary to make that work? For .NET Framework, the OSGroup will always be Windows_NT, right?

```suggestion
  - To run just the new test targetFramework run `dotnet build <Library>.csproj /t:RebuildAndTest /p:BuildTargetFramework=<TargetFramework>`
```

File: docs/coding-guidelines/package-projects.md [hunk 3]
```diff
@@ -94,7 +94,7 @@ Sample \ref .builds file defining a constant used to filter API that were added
   <PropertyGroup>
     <OutputType>Library</OutputType>
     <NuGetTargetMoniker>.NETStandard,Version=v1.7</NuGetTargetMoniker>
-    <DefineConstants Condition="'$(TargetGroup)' == 'netcoreapp1.1'">$(DefineConstants);netcoreapp11</DefineConstants>
+    <DefineConstants Condition="'$(TargetFramework)' == 'netcoreapp1.1'">$(DefineConstants);netcoreapp11</DefineConstants>
```
Comment (ViktorHofer) [inline docs/coding-guidelines/package-projects.md]: This whole document should be updated. Also does the `TargetFramework`  comparison here actually work with the new config system which includes the OSGroup?

File: docs/coding-guidelines/package-projects.md [hunk 4]
```diff
@@ -223,7 +223,7 @@ The primary thing that the library author needs to do in order to ensure the cor
 
 1. Configure the correct projects in your library's `.builds` file.
 2. Reference the `.builds` file from the package project.
-3. Provide a default PackageTargetFramework for empty-TargetGroup builds in the library's `.csproj` or `.vbproj`.
+3. Provide a default PackageTargetFramework for empty-BuildTargetFramework builds in the library's `.csproj` or `.vbproj`.
```
Comment (ViktorHofer) [inline docs/coding-guidelines/package-projects.md]: Are we still using the `PackageTArgetFramework` property?

File: docs/coding-guidelines/project-guidelines.md [hunk 5]
```diff
@@ -6,8 +6,8 @@ once before you can iterate and work on a given library project.
 
 - Setup tools (currently done in restore in build.cmd/sh)
 - Restore external dependencies
- - CoreCLR - Copy to `bin\runtime\$(BuildConfiguration)`
- - Netstandard Library - Copy to `bin\ref\netstandard`
+ - CoreCLR - Copy to `bin\runtime\$(BuildSettings)`
```
Comment (ViktorHofer) [inline docs/coding-guidelines/project-guidelines.md]: BuildSettings?

File: docs/coding-guidelines/project-guidelines.md [hunk 6]
```diff
@@ -17,13 +17,13 @@ once before you can iterate and work on a given library project.
  - Build src\sign.proj
 
 ## Behind the scenes with build-test.cmd/sh
```
Comment (ViktorHofer) [inline docs/coding-guidelines/project-guidelines.md]: Those scripts don't exist anymore, please delete the whole block. Same for build-packages.cmd/sh below.

File: docs/coding-guidelines/updating-ref-source.md [hunk 7]
```diff
@@ -14,5 +14,5 @@ This document provides the steps you need to take to update the reference assemb
 These steps can also be applied to some unique assemblies which depend on changes in System.Private.Corelib coming from [coreclr](https://github.com/dotnet/coreclr) (partial facades like [System.Memory](https://github.com/dotnet/corefx/blob/83711167ee74d2e87cf2d5ed3508c94044bb7edc/src/System.Memory/src/System.Memory.csproj#L6), for example).
```
Comment (ViktorHofer) [inline docs/coding-guidelines/updating-ref-source.md]: removing "comfing from [coreclr]()" and the System.Memory link to corefx.

File: docs/coding-guidelines/updating-ref-source.md [hunk 8]
```diff
@@ -14,5 +14,5 @@ This document provides the steps you need to take to update the reference assemb
...
 1) Build coreclr release.
 2) Build corefx release with coreclr bits (see (// TODO //)) for more details).
-3) Run `msbuild /t:GenerateReferenceSource /p:ConfigurationGroup=Release` from the System.Runtime/ref directory.
+3) Run `msbuild /t:GenerateReferenceSource /p:Configuration=Release` from the System.Runtime/ref directory.
```
Comment (ViktorHofer) [inline docs/coding-guidelines/updating-ref-source.md]: Please update this document by removing 1) and 2) and update 3) to use `dotnet build` instead of `msbuild`.

File: docs/workflow/building/libraries/README.md [hunk 9]
```diff
@@ -77,9 +77,9 @@ The libraries build has two logical components, the native build which produces
 
 The build configurations are generally defaulted based on where you are building (i.e. which OS or which architecture) but we have a few shortcuts for the individual properties that can be passed to the build scripts:
 
-- `-framework|-f` identifies the target framework for the build. It defaults to latest `netcoreapp` but possible values include `netcoreapp` (Which will map to latest netcoreapp) or `net472`. (msbuild property `TargetGroup`)
+- `-framework|-f` identifies the target framework for the build. It defaults to latest `netcoreapp` but possible values include `netcoreapp` (Which will map to latest netcoreapp) or `net472`. (msbuild property `BuildTargetFramework`)
```
Comment (ViktorHofer) [inline docs/workflow/building/libraries/README.md]: "Which will map to latest netcoreapp". This isn't true anymore. Please update that section and mention that currently netcoreapp5.0 needs to be passed in.

File: docs/workflow/building/libraries/README.md [hunk 10]
```diff
@@ -163,14 +163,14 @@ Under the src directory is a set of directories, each of which represents a part
 
 For example the src\libraries\System.Diagnostics.DiagnosticSource directory holds the source code for the System.Diagnostics.DiagnosticSource.dll assembly.
 
-You can build the DLL for System.Diagnostics.DiagnosticSource.dll by going to the `src\libraries\System.Diagnostics.DiagnosticsSource\src` directory and typing `dotnet msbuild`. The DLL ends up in `artifacts\bin\AnyOS.AnyCPU.Debug\System.Diagnostics.DiagnosticSource` as well as `artifacts\bin\runtime\[BuildConfiguration]`.
+You can build the DLL for System.Diagnostics.DiagnosticSource.dll by going to the `src\libraries\System.Diagnostics.DiagnosticsSource\src` directory and typing `dotnet msbuild`. The DLL ends up in `artifacts\bin\AnyOS.AnyCPU.Debug\System.Diagnostics.DiagnosticSource` as well as `artifacts\bin\runtime\[BuildSettings]`.
```
Comment (ViktorHofer) [inline docs/workflow/building/libraries/README.md]: Again, is `BuildSettings` a property defined?

File: docs/workflow/building/libraries/README.md [hunk 11]
```diff
@@ -181,7 +181,7 @@ dotnet msbuild System.Net.NetworkInformation.csproj /p:OSGroup=Linux
 
 - Build release version of library
 ```
-dotnet msbuild System.Net.NetworkInformation.csproj /p:ConfigurationGroup=Release
+dotnet msbuild System.Net.NetworkInformation.csproj /p:Configuration=Release
```
Comment (ViktorHofer) [inline docs/workflow/building/libraries/README.md]:
```suggestion
dotnet build -c Release System.Net.NetworkInformation.csproj
```

File: docs/workflow/building/libraries/README.md [hunk 12]
```diff
@@ -200,7 +200,7 @@ Note that you cannot generally build native components for another OS but you ca
 ### Building in Release or Debug
 
 By default, building from the root or within a project will build the libraries in Debug mode.
-One can build in Debug or Release mode from the root by doing `./build.sh -subsetCategory libraries  -c Release` or `./build.sh -subsetCategory libraries -c Debug` or when building a project by specifying `/p:ConfigurationGroup=[Debug|Release]` after the `dotnet msbuild` command.
+One can build in Debug or Release mode from the root by doing `./build.sh -subsetCategory libraries  -c Release` or `./build.sh -subsetCategory libraries -c Debug` or when building a project by specifying `/p:Configuration=[Debug|Release]` after the `dotnet msbuild` command.
```
Comment (ViktorHofer) [inline docs/workflow/building/libraries/README.md]:
```suggestion
One can build in Debug or Release mode from the root by doing `./build.sh -subsetCategory libraries  -c Release` or `./build.sh -subsetCategory libraries -c Debug`.
```

No need to specify again how to build an individual library with the dotnet CLI as we already showed that above and as we now use the default knobs used in any other customer project: `-c Debug/Release`.

File: docs/workflow/building/libraries/freebsd-instructions.md [hunk 13]
```diff
@@ -66,7 +66,7 @@ index 768b6fb..c02f76f 100644
 +++ b/repos/core-setup.proj
 @@ -4,6 +4,7 @@
    <PropertyGroup>
-     <BuildArguments>-ConfigurationGroup=$(Configuration) -PortableBuild=$(PortableBuild) -SkipTests=true </BuildArguments>
+     <BuildArguments>-Configuration=$(Configuration) -PortableBuild=$(PortableBuild) -SkipTests=true </BuildArguments>
```
Comment (ViktorHofer) [inline docs/workflow/building/libraries/freebsd-instructions.md]: You are changing something in core-setup here. Even though this documented is completely outdated I would revert the change.

File: docs/workflow/debugging/libraries/debugging-packages.md [hunk 14]
```diff
@@ -96,7 +96,7 @@ ResolvePkgProjReferences:
 To run a test from a single Build Pivot combination, specify all properties and build the `csproj`:
 
 ```
-dotnet msbuild System.Net.ServicePoint.Tests.csproj /t:rebuildandtest /p:TargetGroup=netcoreapp2.0 /p:OuterLoop=true /p:xunitoptions=-showprogress /p:ConfigurationGroup=Debug
+dotnet msbuild System.Net.ServicePoint.Tests.csproj /t:rebuildandtest /p:BuildTargetFramework=netcoreapp2.0 /p:OuterLoop=true /p:xunitoptions=-showprogress /p:Configuration=Debug
```
Comment (ViktorHofer) [inline docs/workflow/debugging/libraries/debugging-packages.md]: `dotnet build` again here, and no need to pass in the Configuration here, that's just confusing.

File: eng/Configurations.props [hunk 15]
```diff
@@ -1,7 +1,7 @@
 <Project>
   <!-- Honor the generic RuntimeConfiguration property. -->
   <PropertyGroup>
-    <RuntimeConfiguration Condition="'$(RuntimeConfiguration)' == ''">$(ConfigurationGroup)</RuntimeConfiguration>
+    <RuntimeConfiguration Condition="'$(RuntimeConfiguration)' == ''">$(Configuration)</RuntimeConfiguration>
```
Comment (ViktorHofer) [inline eng/Configurations.props]: cc @jaredpar 

We don't use ConfigurationGroup anymore in libraries. What's the ideal change here?

File: eng/build.ps1 [hunk 16]
```diff
@@ -136,7 +136,7 @@ foreach ($argument in $PSBoundParameters.Keys)
     "build"                { $arguments += " -build" }
     "buildtests"           { if ($build -eq $true) { $arguments += " /p:BuildTests=true" } else { $arguments += " -build /p:BuildTests=only" } }
     "test"                 { $arguments += " -test" }
-    "configuration"        { $configuration = (Get-Culture).TextInfo.ToTitleCase($($PSBoundParameters[$argument])); $arguments += " /p:ConfigurationGroup=$configuration -configuration $configuration" }
+    "configuration"        { $configuration = (Get-Culture).TextInfo.ToTitleCase($($PSBoundParameters[$argument])); $arguments += " /p:Configuration=$configuration -configuration $configuration" }
```
Comment (ViktorHofer) [inline eng/build.ps1]: ConfigurationGroup was passed in as it was necessary before. Now that we are using the standard property we don't need to pass it in additionally anymore.

```suggestion
    "configuration"        { $configuration = (Get-Culture).TextInfo.ToTitleCase($($PSBoundParameters[$argument])); $arguments += " -configuration $configuration" }
```

File: eng/build.sh [hunk 17]
```diff
@@ -92,7 +92,7 @@ while [[ $# > 0 ]]; do
       ;;
      -configuration|-c)
       val="$(tr '[:lower:]' '[:upper:]' <<< ${2:0:1})${2:1}"
-      arguments="$arguments /p:ConfigurationGroup=$val -configuration $val"
+      arguments="$arguments /p:Configuration=$val -configuration $val"
```
Comment (ViktorHofer) [inline eng/build.sh]:
```suggestion
      arguments="$arguments -configuration $val"
```

File: src/libraries/Directory.Build.targets [hunk 18]
```diff
@@ -116,8 +116,8 @@
 
   <Target Name="ReportConfigurationErrorMessage"
           BeforeTargets="AssignProjectConfiguration"
-          Condition="'$(BuildConfigurations)' != ''" >
-    <Message Importance="Low" Text="$(MSBuildProjectFullPath), C: $(Configuration) BC: $(BuildConfiguration) BCs: $(BuildConfigurations)" />
+          Condition="'$(TargetFrameworks)' != ''" >
+    <Message Importance="Low" Text="$(MSBuildProjectFullPath), T: $(TargetFrameowork) BS: $(BuildSettings) Tfms: $(TargetFrameworks)" />
     <Message Importance="High" Condition="'$(ConfigurationErrorMsg)' != ''" Text="$(MSBuildProjectFullPath) ConfigurationErrorMessage: $(ConfigurationErrorMsg)" />
```
Comment (ViktorHofer) [inline src/libraries/Directory.Build.targets]: Is `ConfigurationErrorMsg` still defined in the new configuration system? Looks like this is dead code, I would remove the whole target. Also typo above: `$(TargetFrameowork)`

File: src/libraries/System.Linq/src/System.Linq.csproj [hunk 19]
```diff
@@ -7,7 +7,7 @@
   </PropertyGroup>
   <ItemGroup>
     <ContractProject Include="..\ref\System.Linq.csproj">
-      <TargetGroup>$(NetCoreAppCurrent)</TargetGroup>
+      <BuildTargetFramework>$(NetCoreAppCurrent)</BuildTargetFramework>
```
Comment (ViktorHofer) [inline src/libraries/System.Linq/src/System.Linq.csproj]: cc @ericstj. What is a ContractProject?

File: src/libraries/packages.builds [hunk 20]
```diff
@@ -2,7 +2,7 @@
   <Import Project="$(MSBuildThisFileDirectory)Directory.Build.props" />
 
   <PropertyGroup Condition="'$(DotNetBuildFromSource)' == 'true'">
-    <AdditionalBuildConfigurations>$(AdditionalBuildConfigurations);package-$(Configuration)</AdditionalBuildConfigurations>
+    <AdditionalBuildTargetFrameworks>$(AdditionalAdditionalBuildTargetFrameworks);package-$(Configuration)</AdditionalBuildTargetFrameworks>
```
Comment (ViktorHofer) [inline src/libraries/packages.builds]: This doesn't look right:

1) Typo
2) Was $(Configuration) here refereing to the the expanded string or really just to the configuration string?

cc @ericstj

File: tools-local/scripts/dev/release-2.1-build-deb-rpm-docker.sh [hunk 21]
```diff
@@ -57,7 +57,7 @@ package() {
         Tools/msbuild.sh \
         build.proj \
         /t:BuildTraversalBuildDependencies \
-        /p:ConfigurationGroup=Release \
+        /p:Configuration=Release \
```
Comment (ViktorHofer) [inline tools-local/scripts/dev/release-2.1-build-deb-rpm-docker.sh]: Undo changes in this file, that's core-setup stuff.

File: docs/coding-guidelines/adding-api-guidelines.md (hunk 2 above)
Comment (ericstj) [reply docs/coding-guidelines/adding-api-guidelines.md]: I think explicitly passing in TargetFramework should work (so long as it matches one of the frameworks in TargetFrameworks), shouldn't it @Anipik?  The part that didn't work was the *best* framework detection that used to work before.

File: docs/coding-guidelines/package-projects.md (hunk 3 above)
Comment (ericstj) [reply docs/coding-guidelines/package-projects.md]: Yeah, by the time TargetFramework hits the project, the OSGroup is removed.

File: docs/workflow/debugging/libraries/debugging-packages.md (hunk 14 above)
Comment (ericstj) [reply docs/workflow/debugging/libraries/debugging-packages.md]: Also, can we use TargetFramework and not BuildTargetFramework?

Comment (Anipik) [reply docs/workflow/debugging/libraries/debugging-packages.md]: > Also, can we use TargetFramework and not BuildTargetFramework? 

Do we want people to use TargetFramework here because we will have to specify the targetFramework with the OSGroup as well. We still don't have a check to verify that if a given targetFramework is a supported one.

File: docs/coding-guidelines/adding-api-guidelines.md (hunk 2 above)
Comment (Anipik) [reply docs/coding-guidelines/adding-api-guidelines.md]: yes that should work

File: src/libraries/packages.builds (hunk 20 above)
Comment (Anipik) [reply src/libraries/packages.builds]: It was just the Debug or Release

File: docs/coding-guidelines/adding-api-guidelines.md (hunk 1 above)
Comment (jaredpar) [reply docs/coding-guidelines/adding-api-guidelines.md]: Agree.

---
In reply to: [377837257](https://github.com/dotnet/runtime/pull/32062#discussion_r377837257) [](ancestors = 377837257)

File: eng/Configurations.props (hunk 15 above)
Comment (jaredpar) [reply eng/Configurations.props]: This is the ideal change in the new system: essentially default to `$(Configuration)` when `$(RuntimeConfiguration)` is not specified. 

---
In reply to: [377848745](https://github.com/dotnet/runtime/pull/32062#discussion_r377848745) [](ancestors = 377848745)

File: docs/workflow/building/libraries/README.md (hunk 10 above)
Comment (Anipik) [reply docs/workflow/building/libraries/README.md]: BuildSettings is the new name of buildConfiguration. We have reduced the use of buildConfiguration to just a concatenation of build properties. Coreclr uses buildConfiguration for debug and release. It will help  in having the same properties.
However here we can use the complete exapansion. (We can discuss about the name of the new property but it doesnt matter as much developers wont be using it.)

File: docs/coding-guidelines/package-projects.md (hunk 3 above)
Comment (Anipik) [reply docs/coding-guidelines/package-projects.md]: I dont have full context on different cases we handle in our pkgproj. so i am leaving this one to another PR. I will probably talk with someone before updating the docs

**Comment by Anipik** [comment]
@ericstj @ViktorHofer can you take another go at this ?

File: docs/coding-guidelines/adding-api-guidelines.md [hunk 22]
```diff
@@ -24,42 +24,21 @@ the implementation without compat concerns in future releases.
...
   - If assembly or package version is updated the package index needs to be updated by running
     `dotnet msbuild <Library>/pkg/<Library>.pkgproj /t:UpdatePackageIndex`
 
 **Update tests**
-  - Set `TargetGroup` which will generally match the `TargetGroup` in the src library build configuration. (ex: [System.Runtime\tests\Configurations.props](https://github.com/dotnet/corefx/blob/master/src/System.Runtime/tests/Configurations.props#L3))
+  - Add new `TargetFramework` to the ```TargetFrameworks```.
   - Add new test code following [conventions](project-guidelines.md#code-file-naming-conventions) for new files to that are specific to the new target framework.
-  - To run just the new test configuration run `dotnet msbuild <Library>.csproj /t:RebuildAndTest /p:TargetGroup=<TargetGroup>`
+  - To run just the new test targetFramework run `dotnet build <Library>.csproj /t:RebuildAndTest /p:TargetFramework=<TargetFramework>`. TargetFramework should be chosen only from supported TargetFrameworks.
```
Comment (ViktorHofer) [inline docs/coding-guidelines/adding-api-guidelines.md]: nit: can we use `-f` instead of using `/p:TargetFramework`? We should start using the default CLI switches like `-c`, `-f` and others.

File: docs/workflow/debugging/libraries/debugging-packages.md [hunk 23]
```diff
@@ -96,7 +96,7 @@ ResolvePkgProjReferences:
 To run a test from a single Build Pivot combination, specify all properties and build the `csproj`:
 
 ```
-dotnet msbuild System.Net.ServicePoint.Tests.csproj /t:rebuildandtest /p:TargetGroup=netcoreapp2.0 /p:OuterLoop=true /p:xunitoptions=-showprogress /p:ConfigurationGroup=Debug
+dotnet msbuild System.Net.ServicePoint.Tests.csproj /t:rebuildandtest /p:TargetFramework=netcoreapp2.0 /p:OuterLoop=true /p:xunitoptions=-showprogress
```
Comment (ViktorHofer) [inline docs/workflow/debugging/libraries/debugging-packages.md]:
```suggestion
dotnet build System.Net.ServicePoint.Tests.csproj -f netcoreapp2.0 /t:rebuildandtest /p:OuterLoop=true /p:xunitoptions=-showprogress
```

File: docs/workflow/testing/libraries/testing.md [hunk 24]
```diff
@@ -50,9 +50,9 @@ dotnet msbuild /t:RebuildAndTest /p:XunitMethodName={FullyQualifiedNamespace}.{C
 
 #### Running tests in a different target framework
 
-Each test project can potentially have multiple build configurations. There are some tests that might be OS-specific, or might be testing an API that is available only on some target frameworks, so the `BuildConfigurations` property specifies the valid configurations. By default we will build and run only the default build configuration which is `netcoreapp`. The rest of the configurations will need to be built and ran by specifying the configuration options.
+Each test project can potentially have multiple target frameworks. There are some tests that might be OS-specific, or might be testing an API that is available only on some target frameworks, so the `TargetFrameworks` property specifies the valid target frameworks. By default we will build and run only the default build target framework which is `netcoreapp5.0`. The rest of the targetframeworks will need to be built and ran by specifying the BuildTargetFramework options.
 
 ```cmd
 cd src\libraries\System.Runtime\tests
-dotnet msbuild System.Runtime.Tests.csproj /p:TargetGroup=net472
+dotnet msbuild System.Runtime.Tests.csproj /p:BuildTargetFramework=net472
```
Comment (ViktorHofer) [inline docs/workflow/testing/libraries/testing.md]:
```suggestion
dotnet build System.Runtime.Tests.csproj -f net472
```

File: eng/build.ps1 [hunk 25]
```diff
@@ -136,7 +136,7 @@ foreach ($argument in $PSBoundParameters.Keys)
     "build"                { $arguments += " -build" }
     "buildtests"           { if ($build -eq $true) { $arguments += " /p:BuildTests=true" } else { $arguments += " -build /p:BuildTests=only" } }
     "test"                 { $arguments += " -test" }
-    "configuration"        { $configuration = (Get-Culture).TextInfo.ToTitleCase($($PSBoundParameters[$argument])); $arguments += " /p:ConfigurationGroup=$configuration -configuration $configuration" }
+    "configuration"        { $configuration = (Get-Culture).TextInfo.ToTitleCase($($PSBoundParameters[$argument])); $arguments += " -configuration $configuration" }
```
Comment (ViktorHofer) [inline eng/build.ps1]: Remove the $configuration property as it isn't used anymore:

```suggestion
    "configuration"        { $arguments += " -configuration $((Get-Culture).TextInfo.ToTitleCase($($PSBoundParameters[$argument])))" }
```

**Review by ViktorHofer — APPROVED** [review]
A few suggestions and the question about `ContractProject` in System.Linq.csproj.
